### PR TITLE
API to upload token metadata JSON to IPFS

### LINKED ISSUE #5: Endpoint for storing json token metadata on ipfs
JSON object should have the following values.

```javascript
{
"name": "3D Cryptobot",
"symbol": "CB",
"decimals": 0,
"description": "Custom made NFT 3d cryptobot",
"isBooleanAmount": true,
"isTransferable": true,
"shouldPreferSymbol": false,
"language": "en",
"tags": ["3D model", "Collectables", "NFT", "Cryptobot"],
"attributes": [  {
    "color_schema": {"": ""},
    "body_parts_type": {"": ""} 
  }  ], 
"externalUri": "https://cryptocodeschool.in/tezos/",
"artifactUri": "<insert-3d-model-ipfs-url-here>",
"displayUri": "<insert-3d-model-ipfs-image-url-here>"
}
```

### PR DESCRIPTION
**Pull request by manangouhari**
**What does this PR do?**
Implements the API which uploads token metadata to IPFS.
Takes this json body as input - 

```
{
  "artifactURI": <ipfs_hash_of_3d_model>,
  "displayURI": <ipfs_hash_of_image>
}
```

and pins it to IPFS using Pinata.

Question for @bhaskarSingh :
Currently it embeds just the IPFS hash in the metadata. 
Should it embed the whole IPFS url like - `https://cloudflare-ipfs/ipfs/<hash>` or is just the hash enough?

### CODE REVIEW

**Comment by bhaskarSingh** [comment]
> Should it embed the whole IPFS url like - https://cloudflare-ipfs/ipfs/<hash> or is just the hash enough?

Storing just the hash makes sense. eg: `ipfs://<hash>`. This allows using a different gateway later if required.

**Comment by bhaskarSingh** [comment]
For now, remove the `attributes` key-value pair.

**Comment by manangouhari** [comment]
> Storing just the hash makes sense. eg: ipfs://<hash>. This allows using a different gateway later if required.

Done.


> For now, remove the attributes key-value pair.

Done.

**Comment by manangouhari** [comment]
Now, the updated JSON parameter is - 
```
{
  "artifactURI": <ipfs_hash_of_3d_model>,
  "displayURI": <ipfs_hash_of_image>,
  "tokenID": <token_id>
}

```

**Review by bhaskarSingh — APPROVED** [review]
Looks great. I haven't tested it manually. Please do a self-review and merge it if everything looks alright.

File: app.js [hunk 1]
```diff
@@ -49,10 +49,47 @@ app.use(express.json());
 // logging middleware
 app.use(morgan("tiny"));
 
+app.post("/api/upload-json-metadata-to-ipfs", async (req, res) => {
```
Comment (bhaskarSingh) [inline app.js:52]: @manangouhari  Just need a tiny update. Please wrap it in try & catch block and please implement a proper edge case for handling error.

Comment (manangouhari) [reply app.js:52]: Handled here - #7

**Pull request merged**